benchmarks/benchmark/tools/locust-load-inference/locust-docker/locust-tasks/custom_metric_aggregator.py

```diff
@@ -8,28 +8,56 @@ def __init__(self):
         self.test_time = []
         self.success_count = 0
         self.failure_count = 0
-        
 
-    def add_metric(self, sent, received, test_time, request_successful_bool):
-        if request_successful_bool==1:
+    def add_metric(self, sent, received, test_time, request_succesful_bool):
+        if request_succesful_bool == 1:
             self.tokens_sent.append(sent)
             self.tokens_received.append(received)
             self.test_time.append(test_time)
             self.success_count += 1
         else:
             self.failure_count += 1
-            
+
+    def add_metrics(self, tokens_sent, tokens_received, test_time, success_count, failure_count):
+        self.tokens_sent = self.tokens_sent + tokens_sent
+        self.tokens_received = self.tokens_received + tokens_received
+        self.test_time = self.test_time + test_time
+        self.success_count += success_count
+        self.failure_count += failure_count
+
+    def share_stats(self):
+        return self.tokens_sent, self.tokens_received, self.test_time, self.success_count, self.failure_count
 
     def calculate_average_tokens(self):
-        avg_sent = sum(self.tokens_sent) / len(self.tokens_sent) if self.tokens_sent else 0
-        avg_received = sum(self.tokens_received) / len(self.tokens_received) if self.tokens_received else 0
-        avg_test_time = sum(self.test_time) / len(self.test_time) if self.tokens_sent else 0
-        
-        return avg_sent, avg_received, avg_test_time
+        if self.tokens_sent and len(self.tokens_sent) > 0:
+            avg_sent = sum(self.tokens_sent) / \
+                len(self.tokens_sent) if self.tokens_sent else 0
+            avg_received = sum(self.tokens_received) / \
+                len(self.tokens_received) if self.tokens_received else 0
+            avg_test_time = sum(self.test_time) / \
+                len(self.test_time) if self.tokens_sent else 0
+            avg_output_token_latency = 0
+            for i in range(0, len(self.tokens_received)):
```

Comment:
Shouldn't we be loop through the total requests instead of tokens_received for calculating output token latency per request?

Reply:
I can change range(tokens_received) to use range(success_count) if preferred as that should have the same value, and is likely more intuitive? Does that cover the question here?

Reply:
I'm assuming success_count is the total # of requests which succeeded? In that case, it makes sense to use that.

Reply:
Changed!